borg/archive.py

```diff
@@ -263,8 +265,9 @@ def save(self, name=None, timestamp=None):
             'time': start.isoformat(),
             'time_end': end.isoformat(),
             'chunker_params': self.chunker_params,
-        })
-        data = msgpack.packb(metadata, unicode_errors='surrogateescape')
+        }
+        metadata.update(additional_metadata or {})
+        data = msgpack.packb(StableDict(metadata), unicode_errors='surrogateescape')
```

Comment:
That's a good idea. I'm not familiar with the details of this pull request so it's better that you handle the conflicts.